guides/assets/javascripts/guides.js

```diff
@@ -102,5 +103,91 @@
       }, 3000);
       e.clearSelection();
     });
+
+    var mainColElems = Array.from(document.getElementById("mainCol").children);
+    var navLinks = document.querySelectorAll("#subCol a");
+    var DESKTOP_THRESHOLD = 1024;
+
+    var matchingNavLink = function (elem) {
+      if(!elem) return;
+      var index = mainColElems.indexOf(elem);
+
+      var match;
+      while (index >= 0 && !match) {
+        var link = mainColElems[index].querySelector("a");
+        if (link) {
+          match = document.querySelector('#subCol [href="' + link.getAttribute("href") + '"]');
+        }
+        index--;
+      }
+      return match;
+    }
+
+    var removeHighlight = function () {
+      for (var i = 0, n = navLinks.length; i < n; i++) {
+        navLinks[i].classList.remove("active");
+      }
+    }
+
+    var updateHighlight = function (elem) {
+      removeHighlight();
+      if (!elem) return;
+      elem.classList.add("active");
+      if (window.innerWidth > DESKTOP_THRESHOLD) {
```

Comment:
I see this is to scroll the nav to the right position, but should we only initialize all this highlight logic for desktop, or rather when the navigation is on the side? It doesn't make much sense to trigger it on mobile where the nav is on top of the content, does it?

No big deal to trigger it there, it's just code we may not need to run on those devices.

Reply:
I was thinking if someone has a narrow window and then resizes to a wider one maybe the highlight should be there on resize. But it would appear once they scroll in that case so I'll remove it.